.github/actions/deploy/deploy.py

```
@@ -0,0 +1,498 @@
+
+        # Create temp directory for operations
+        self.temp_dir = tempfile.mkdtemp()
+        print(f'📁 Working directory: {self.temp_dir}')
+
+        # Set deployment namespace from input args
+        self.deployment_namespace = self.args.namespace
+        print(f'🏷️  Deployment namespace: {self.deployment_namespace}')
+
+    def run_command(self,
+                    cmd: List[str],
+                    cwd: str = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)